### PR TITLE
"Create folder" button for save dialog

### PR DESCRIPTION
**Pull request by wolfreak99**
"platform_ensure_directory_exists" forces a directory to be created and i realized that if someone wants to create a folder, they need to use the system dialog window, which will disconnect any network games they are in.

### CODE REVIEW

File: data/language/en-GB.txt [hunk 1]
```diff
@@ -4232,6 +4232,9 @@ STR_5920    :Render weather effects
 STR_5921    :{SMALLFONT}{BLACK}If enabled, rain and gloomy colours will be rendered during storms.
 STR_5922    :{POP16}{POP16}{POP16}{POP16}{POP16}{POP16}{SMALLFONT}{BLACK}Max {STRINGID}
 STR_5923    :{POP16}{POP16}{POP16}{POP16}{POP16}{POP16}{POP16}{POP16}{SMALLFONT}{BLACK}Max {COMMA16} {STRINGID} per train
+STR_5924    :New folder
```
Comment (janisozaur) [inline data/language/en-GB.txt]: this should be named "directory"

File: src/windows/loadsave.c [hunk 2]
```diff
@@ -38,27 +38,29 @@ enum {
 	WIDX_BACKGROUND,
 	WIDX_TITLE,
 	WIDX_CLOSE,
+	WIDX_DEFAULT,
 	WIDX_UP,
-	WIDX_NEW,
+	WIDX_NEW_FOLDER,
```
Comment (janisozaur) [inline src/windows/loadsave.c]: `WIDX_NEW_DIRECTORY`

File: data/language/en-GB.txt (hunk 1 above)
Comment (AaronVanGeffen) [inline data/language/en-GB.txt]: Both Windows and macOS use 'folder' to describe directories to the user. Actually, even KDE and GNOME do, too. So, it makes sense to use use 'folder', really.

Comment (janisozaur) [inline data/language/en-GB.txt]: `$CWD`, `cd`, `dir`, and just a handful of ["folder"](https://github.com/torvalds/linux/search?utf8=%E2%9C%93&q=folder) vs ["directory"](https://github.com/torvalds/linux/search?utf8=%E2%9C%93&q=directory) in public *filesystem* sources all point to "directory".

Comment (AaronVanGeffen) [inline data/language/en-GB.txt]: Command line tools versus graphical user interface. I'm not disputing the use of 'directory' internally -- it is the technical term for it, no doubt.

The file / folder analogy is used consistently in GUIs though -- not just in strings, but also in icons. I do believe we should follow that -- the consistency argument works that way, too.

Comment (IntelOrca) [inline data/language/en-GB.txt]: @janisozaur Typically they are refereed to as directories on a command line and folders in graphical user interfaces.

https://en.wikipedia.org/wiki/Directory_(computing)#Folder_metaphor

File: data/language/en-GB.txt [hunk 3]
```diff
@@ -4232,6 +4232,9 @@ STR_5920    :Render weather effects
...
+STR_5925    :Type the name of the folder.
```
Comment (Gymnasiast) [inline data/language/en-GB.txt]: I would change this to `Type the name of the folder.`

File: src/windows/loadsave.c [hunk 4]
```diff
@@ -389,43 +394,71 @@ static void window_loadsave_scrollmouseover(rct_window *w, int scrollIndex, int
 
 static void window_loadsave_textinput(rct_window *w, int widgetIndex, char *text)
 {
+	if (text == NULL || text[0] == 0)
```
Comment (Gymnasiast) [inline src/windows/loadsave.c]: Could you move this under the declaration again? It's convention to have these either locally or at the top of the function.

File: src/windows/loadsave.c [hunk 5]
```diff
@@ -389,43 +394,71 @@ static void window_loadsave_scrollmouseover(rct_window *w, int scrollIndex, int
...
-		else {
-			window_error_open(STR_ERROR_INVALID_CHARACTERS, STR_NONE);
-		}
-		return;
-	}
 
-	safe_strcpy(path, _directory, sizeof(path));
-	safe_strcat_path(path, text, sizeof(path));
-	path_append_extension(path, _extension, sizeof(path));
+		safe_strcpy(path, _directory, sizeof(path));
```
Comment (Gymnasiast) [inline src/windows/loadsave.c]: These three lines need extra indentation.

**Comment by Gymnasiast** [comment]
I left a few remarks, but it otherwise looks good codewise. When you have changed the bits, could you rebase it? After that I'll test it.

**Comment by wolfreak99** [comment]
@Gymnasiast  I fixed the remarks and rebased it, i havent had a chance yet to test it but i'm about to go into work, if you can test compilation before i get home (in about 8 hours from the time this is posted) cool, if not, i'll test it and let you know you can test it after i get home.

File: src/windows/loadsave.c [hunk 6]
```diff
@@ -547,12 +580,13 @@ static void window_loadsave_populate_list(rct_window *w, int includeNewItem, con
 	_listItemsCount = 0;
 
 	// Show "new" button when saving
```
Comment (Gymnasiast) [inline src/windows/loadsave.c]: ```Show "new" buttons when saving```

**Comment by Gymnasiast** [comment]
Hm, it works, but I feel the toolbar is getting a bit cluttered. I think the window would benefit from using icon rather than text. The problem is that I don't think we have suitable sprites at the moment.
Any ideas, @IntelOrca @duncanspumpkin @Broxzier ?

**Comment by IntelOrca** [comment]
@Gymnasiast I don't have any ideas

**Comment by Gymnasiast** [comment]
Hm, ok. Well, I'm not going to hold back this PR for it, but I will create an issue for it.
@wolfreak99 If you could address the two comments I left a line note about, this PR will be merged.

**Comment by Broxzier** [comment]
@Gymnasiast: I agree that the save game dialog could use some work, but this PR is in line with what is already there. Tidying it up is another task.

**Comment by wolfreak99** [comment]
@Gymnasiast  I have fixed the comments. I have also merged it all down to one commit as to not clog the history with "Formatting improvements" etc.

**Pull request merged**